.github/workflows/changeset-check.yml

```
@@ -0,0 +1,64 @@
+        with:
+          node-version: '23' # Match the version used in typescript-publish.yml
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 10 # Match user's pnpm version
+          run_install: false # We don't need to install root dependencies for this check
+
+      # Need to determine which paths were changed to conditionally run the checks
+      # This uses the dorny/paths-filter action to check changed files
+      - uses: dorny/paths-filter@v2
```

actionlint (1.7.4)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)